index.js

```diff
@@ -126,7 +126,9 @@ function replacePath (path, keys) {
       name: key || index++,
       delimiter: prefix || '/',
       optional: optional,
-      repeat: repeat
+      repeat: repeat,
+      offset: i + (prefix ? prefix.length : 0),
```

Comment:
I'll PR my code too for you to review. There's a little more that needs to be added to support this, but I think it was a nifty start. I added a "pattern" property (we can validate the user input against it) and the "prefix" property (it's like delimiter, but sometimes there's no prefix and we need that knowledge).